graphql_compiler/compiler/ir_lowering_sql/__init__.py

```diff
@@ -64,12 +69,15 @@ def lower_ir(ir_blocks, query_metadata_table, type_equivalence_hints=None):
                         block, tree_root, ir_blocks, query_metadata_table))
             tree_root = SqlNode(block=block, query_path=query_path)
             query_path_to_node[query_path] = tree_root
+        elif isinstance(block, blocks.Filter):
+            query_path_to_filters.setdefault(query_path, []).append((block, query_path))
```

Comment:
I'm curious why append `(block, query_path)` when `query_path` is already the key that corresponds to the list of those values. That means that for a given `query_path` value of `q`, the value entry for `q` looks like `[(b1, q), (b2, q), (b3, q)]`. I don't see the value of including the exact same query path as the second value of every tuple for each key.

Reply:
This was important when it came time to join other tables. In that case what I had been doing was collapsing filters onto the root node, in which case each filter had to know the query path it applied to. That's a little convoluted so thinking about it now, when we do the joins, we'll keep track of the visited query paths, and then when it comes time to do filters we'll collect the filters at all visited query paths. That should be a little cleaner and easier to understand, and then things aren't moving around as much.